assets/js/dashboard/stats/reports/list.js

```diff
@@ -139,6 +140,20 @@ export default function ListReport(props) {
     return () => { document.removeEventListener('tick', fetchData) }
   }, [props.keyLabel, props.query, visible]);
 
+  // returns a filtered `metrics` list. Since currently, the backend can return different
+  // metrics based on filters and existing data, this function validates that the metrics
+  // we want to display are actually there in the API response.
+  function validatedMetrics() {
+    return metrics.filter((metric) => {
+      return state.list.some((listItem) => listItem[metric.name] != null)
+    })
+  }
+
+  function hiddenOnMobileClass(i) {
+    if (i > MOBILE_MAX_METRIC_COLS) { return 'hidden md:block'}
+    return ''
```

Comment:
Feel free to ignore, but I think the early return in this case makes it harder to read.

```suggestion
    if (i > MOBILE_MAX_METRIC_COLS) {
      return 'hidden md:block'
    } else {
      return ''
    }
```

You could also classNames and format the whole class name (including `text-right`):

```js
return classNames('text-right', {'hidden md:block': i > MOBILE_MAX_METRIC_COLS})
```

Reply:
Sure, a standard if-else block probably reads easier indeed (✅ in https://github.com/plausible/analytics/pull/3209/commits/8ba23b25779e12388225c851c9ad44c7f6e0e984). But I'd prefer to keep it in a separate function that describes the problem that we're solving so that it's clear that this logic is required in two places (column headers and column body).

```js
classNames('text-right', {'hidden md:block': i > MOBILE_MAX_METRIC_COLS})
```

would be nice, but I can't think of a good function name in this scenario 😃